experimental/language-generation/docs/Functions-injected-from-LG.md

```diff
@@ -0,0 +1,222 @@
+# Functions injected from LG library
+
+## Functions
+- [ActivityAttachment](#ActivityAttachment)
+- [template](#template)
+- [fromFile](#fromFile)
+- [isTemplate](#isTemplate)
+
+<a name="ActivityAttachment"></a>
+### ActivityAttachment
+
+Return an activityAttachment from objects.
+
+```
+ActivityAttachment(<collection-of-objects>)
+```
+
+| Parameter | Required | Type | Description |
+| --------- | -------- | ---- | ----------- |
+| <*collection-of-objects*> | Yes | Array or object  | A collection of objects|
+|||||
+
+| Return value | Type | Description |
+| ------------ | -----| ----------- |
+| <*activityAttachment*> | Object | An activityAttachment formed from the input |
+||||
+
+*Example*
+
+This example converts an collection of objects to an activityAttachment.
+Suppose we want to convert a JSON file to herocard attachment. 
+the content in the herocard.json: 
+
+```
+{
+  "title": "titleContent",
+  "text": "textContent",
+  "Buttons": [
+    {
+      "type": "imBack",
+      "Title": "titleContent",
+      "Value": "textContent",
+      "Text": "textContent"
+    }
+  ],
+  "tap": {
+    "type": "@{type}",
+    "title": "@{title}",
+    "text": "@{title}",
+    "value": "@{value}"
+  }
+}
+```
+
+By calling ActivityAttachment in a template, type, title, value are passed from template name.
+
+```
+# externalHeroCardActivity(type, title, value)
+[Activity
+    attachments = @{ActivityAttachment(json(fromFile('.\\herocard.json')), 'herocard')}
+]
+```
+
+And returns a herocard:
+
+```
+{
+    "lgType" = "attachment",
+    "contenttype" = "herocard",
+    "content" = {
+        "title": "titleContent",
+        "text": "textContent",
+        "Buttons": [
+            {
+            "type": "imBack",
+            "Title": "titleContent",
+            "Value": "textContent",
+            "Text": "textContent"
+            }
+        ],
+        "tap": {
+            "type": "@{type}",
+            "title": "@{title}",
+            "text": "@{title}",
+            "value": "@{value}"
+        }
+    }
+}
+```
+
+<a name="template"></a>
+### template
+
+Return the evaluated result of the template name and parameters.
+
+```
+template(<collection-of-objects>)
+```
+
+| Parameter | Required | Type | Description |
+| --------- | -------- | ---- | ----------- |
+| <*collection-of-objects*> | Yes | Array or object  | A collection of objects|
```

Comment:
function `template` has two parameters, the first parameter is templateName(string), and second parameter is scope (object)